share/wake/lib/core/regexp.wake

```diff
@@ -81,6 +81,28 @@ export def regExpToString (regExp: RegExp): String =
 export def matches (testRegExp: RegExp) (str: String): Boolean =
     (\_ \_ prim "match") testRegExp str
 
+# Test if a String matches all provided regular expressions
+# matchesAll (`a*`, `*b`) "ab" = True
+# matchesAll (`a*`, `*b`) "a" = False
+export def matchesAll (testRegExps: List RegExp) (str: String): Boolean = match testRegExps
+    Nil = True
+    head, tail =
+        if (matches head str) then
+            matchesAll tail str
+        else
+            False
```

Comment:
This is too generic to be in regex, use forall/exist in list.wake

Reply:
ah ye. makes sense. fixed!